mailbox/cassandra/src/main/java/org/apache/james/mailbox/cassandra/mail/CassandraAnnotationMapper.java

```diff
@@ -47,12 +46,17 @@ public CassandraAnnotationMapper(CassandraId mailboxId, Session session) {
         this.session = session;
     }
 
-    public List<MailboxAnnotation> getAnnotations() {
-        return getResult(getStoredAnnotationRow());
+    public List<MailboxAnnotation> getAllAnnotations() {
+        return getStoredAnnotationRow()
+                .map(this::toAnnotation)
+                .collect(Collectors.toList());
     }
 
-    public List<MailboxAnnotation> getAnnotations(List<String> keys) {
-        return getResult(getStoredAnnotationRow(keys));
+    public List<MailboxAnnotation> getAnnotationsByKeys(List<String> keys) {
+        return getStoredAnnotationRow()
+                .filter(row -> keys.contains(row.getString(CassandraAnnotationTable.KEY)))
```

Comment:
It's not I wanted you to do, I'm sorry if I wasn't clear enough. You should build the query, then add a filter clause on the query, then run it.
Here, you build the query, run it, then filter, so you get too many results from the DB.